rust-runtime/aws-smithy-http-client/Cargo.toml

```diff
@@ -83,6 +85,8 @@ http-1x = { package = "http", version = "1" , optional = true }
 http-body-1x = { package = "http-body", version = "1", optional = true}
 hyper-rustls = { version = "0.27", features = ["http2", "http1", "native-tokio", "tls12"], default-features = false, optional = true }
 rustls = { version = "0.23", default-features = false, optional = true }
+s2n-tls-hyper = { version = "0.0.1", optional = true }
```

Comment:
smithy-rs is s2n-tls-hyper's first integration. As such, we've released an unstable 0.0.1 version for the purposes of this PR, in case smithy-rs requests that a breaking change be made to s2n-tls-hyper. After this PR is approved, I will release a stable version and update the version here.

Reply:
Since we don't expose s2n-tls-hyper directly then you can probably leave it unstable until you feel ready to stabilize the API (`rustls`, `hyper-rustls`, and `hyper-util` are all unstable as well so not really a problem).